File: apps/api/app/schemas/clinic.py
"""Schemas for the clinic settings (read-only) view."""

from __future__ import annotations

from uuid import UUID

from pydantic import BaseModel


class ClinicMemberOut(BaseModel):
    email: str
    full_name: str
    role: str


class ClinicSettingsResponse(BaseModel):
    clinic_id: UUID
    clinic_name: str
    your_role: str
    members: list[ClinicMemberOut]

File: apps/web/lib/clinic.ts
import { apiFetch } from "@/lib/api";

export interface ClinicMember {
  email: string;
  full_name: string;
  role: string;
}

export interface ClinicSettings {
  clinic_id: string;
  clinic_name: string;
  your_role: string;
  members: ClinicMember[];
}

export async function getClinicSettings(): Promise<ClinicSettings | null> {
  const resp = await apiFetch("/api/clinic/settings");
  if (!resp.ok) return null;
  return (await resp.json()) as ClinicSettings;
}
